mlflow/server/js/src/components/artifact-view-components/ShowArtifactMapView.js

```diff
@@ -0,0 +1,132 @@
+    error: undefined,
+    features: undefined,
+  };
+
+  componentDidMount() {
+    this.fetchArtifacts();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.path !== prevProps.path || this.props.runUuid !== prevProps.runUuid) {
+      this.fetchArtifacts();
+    }
+
+    if (this.leafletMap !== undefined) {
+      if (this.leafletMap.hasOwnProperty('_layers')) {
+        this.leafletMap.off();
+        this.leafletMap.remove();
+        document.getElementsByClassName('map-container')[0].innerHTML = "<div id='map'></div>";
+        this.leafletMap = undefined;
+      }
+    }
+
+    if (this.state.features !== undefined) {
+      const map = L.map('map');
+
+      // Load tiles from OSM with the corresponding attribution
+      // Potentially, these could be set in an ENV VAR to use a custom map
+      const tilesURL = 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+      const attr = '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors';
+
+      L.tileLayer(tilesURL, {
+        attribution: attr,
+      }).addTo(map);
+
+      const geojsonLayer = L.geoJSON(this.state.features, {
+        style(feature) {
+          return feature.properties && feature.properties.style;
+        },
+        pointToLayer(feature, latlng) {
+          if (feature.properties && feature.properties.style) {
+            return L.circleMarker(latlng, feature.properties && feature.properties.style);
+          } else if (feature.properties && feature.properties.icon) {
+            return L.marker(latlng, {
+              icon: L.icon(feature.properties && feature.properties.icon),
+            });
+          }
+          return L.marker(latlng, {
+            icon: L.icon({
+              iconRetinaUrl: iconRetina,
+              iconUrl: icon,
+              shadowUrl: iconShadow,
+              iconSize: [24, 36],
+              iconAnchor: [12, 36],
+            }),
+          });
+        },
+        onEachFeature: onEachFeature,
+      }).addTo(map);
+      map.fitBounds(geojsonLayer.getBounds());
+      this.leafletMap = map;
+    }
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <div>Loading...</div>;
+    }
+    if (this.state.error) {
+      return <div>Oops we couldn't load your file because of an error.</div>;
+    } else {
+      return (
+        <div className='map-container'>
+          <div id='map'></div>
+        </div>
+      );
+    }
+  }
+
+  fetchArtifacts() {
+    const getArtifactRequest = new Request(getSrc(this.props.path, this.props.runUuid), {
+      method: 'GET',
+      redirect: 'follow',
+      headers: new Headers(getRequestHeaders(document.cookie)),
+    });
+    fetch(getArtifactRequest)
+      .then((response) => {
+        return response.blob();
+      })
+      .then((blob) => {
+        const fileReader = new FileReader();
+        fileReader.onload = (event) => {
+          this.setState({ features: JSON.parse(event.target.result), loading: false });
```

Comment:
We should catch errors while parsing the GeoJSON artifact, e.g. if the user logs a malformed artifact:

```suggestion
try {
  this.setState({ features: JSON.parse(event.target.result), loading: false });
}
catch(error) {
  this.setState({ error, loading: false, features: undefined }
}
```

Reply:
Isn't line 128 doing that already?

Reply:
Good question - the `catch` statement below ends up not applying here, since it doesn't catch errors thrown in the `filereader.onload` callback (you can reproduce the behavior I'm referring to by logging a malformed geojson file & attempting to view it in the UI).